src/main/webapp/script.js

```diff
@@ -742,22 +830,40 @@ function searchToken(cy, query) {
 }
 
 /**
- * Zoom/highlights node based on queried token and index
+ * Zoom/highlights node at given index in array
  *
  * @param cy the graph to search through
  * @param nodesWithToken the list of nodes that contain specified token
- * @param num the target index of the list of nodes that contain token
- * @returns the result of the search.
+ * @param num the target index
  */
 function updateHighlightedToken(cy, nodesWithToken, num) {
-  if (num < 0) {
+  // Prevent the user from navigating to a node before the first node
+  // Also, if we are at the first node, disable the previous button
+  // so the user can't go back
+  if (num <= 0) {
     num = 0;
+    document.getElementById('prevnode').disabled = true;
+  } else {
+    document.getElementById('prevnode').disabled = false;
   }
-  if (num > nodesWithToken.length - 1) {
+  // Prevent the user from navigating to a node after the last node
+  // Also, if we are at the last node, disable the next button
+  // so the user can't go forward
+  if (num >= nodesWithToken.length - 1) {
     num = nodesWithToken.length - 1;
+    document.getElementById('nextnode').disabled = true;
+  } else {
+    document.getElementById('nextnode').disabled = false;
   }
+
+  // Remove highlight from all nodes and associated edges
```

Comment:
This removes the highlighting from the other searched nodes while focusing only on one of them, right? I actually think it's useful to highlight all of the searched nodes (if I choose to drag the graph around to see them), and clicking "prevnode" or "nextnode" should only help me zoom in on the node for clarity, but not remove the highlighting.

Reply:
Should we still only show the edges of the zoomed-in node in dashed form?

Reply:
yeah, my comment was mainly about node-highlighting